Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+
+            // If there is associated event configuration for the state, then it is valid
+            if (eventConfigType != null)
+            {
+                return true;
+            }
+            // If an associated class for the state could not be found then the event config is not valid
+            else
+            {
+                return false;
+            }
+        }
+
+        // Create an instance of an associated event scriptable object given the state.
+        private void CreateEventScriptable(SerializedProperty eventConfiguration, string stateName)
+        {
+            string className = stateName + "InteractionEventConfiguration";
+
+            // If the event configuration for the state is currently, then create a scrpitable instance
+            if (eventConfiguration.objectReferenceValue == null)
+            {
+                // Initialize the associated scriptable object event configuration with the correct state 
+                eventConfiguration.objectReferenceValue = ScriptableObject.CreateInstance(className);
+                eventConfiguration.objectReferenceValue.name = stateName + "EventConfiguration";
+            }
+        }
+
+        // Draw custom container for the event configuration scriptable 
+        private void DrawStateEventsScriptableSubEditor(SerializedProperty scriptable)
+        {
+            if (scriptable.objectReferenceValue != null)
+            {
+                UnityEditor.Editor configEditor = UnityEditor.Editor.CreateEditor(scriptable.objectReferenceValue);
+                
+                using (new EditorGUILayout.VerticalScope(InspectorUIUtility.Box(10)))
+                {
+                    EditorGUILayout.Space();
+                    configEditor.OnInspectorGUI();
+                    EditorGUILayout.Space();
+                }
+            }
+        }
+
+        private void SetCoreStateType(SerializedProperty stateProp, SerializedProperty stateNameProp)
+        {
+            Rect position = EditorGUILayout.GetControlRect();
+            using (new EditorGUI.PropertyScope(position, new GUIContent("State"), stateProp))
+            {
+                string[] coreInteractionStateNames = Enum.GetNames(typeof(CoreInteractionState)).ToArray();
+                int id = Array.IndexOf(coreInteractionStateNames, -1);
+                int newId = EditorGUI.Popup(position, id, coreInteractionStateNames);
+
+                // NOTE FOR THE FUTURE: Sort the core states in a menu that indicates whether a core state is a near 
+                // interaction state, far interaction state, or both to futher push the mental model of the MRTK interaction model
+
+                if (newId != -1)
+                {
+                    string selectedState = coreInteractionStateNames[newId];
+
+                    // If this state is not already being tracked then change the name
+                    if (!instance.IsStateTracked(selectedState))
```

Comment:
Could you instead only show in the dropdown the list of states that are not already tracked?